alacritty_terminal/src/term/mod.rs

```diff
@@ -2118,6 +2123,36 @@ mod tests {
         assert_eq!(term.grid()[cursor].c, '▒');
     }
 
+    #[test]
+    fn clear_viewport_updates_vi_mode_cursor_pos() {
+        let size = SizeInfo::new(10.0, 20.0, 1.0, 1.0, 0.0, 0.0, false);
+        let mut term = Term::new(&MockConfig::default(), size, ());
+
+        // Create 10 lines of scrollback.
+        for _ in 0..29 {
+            term.newline();
+        }
+
+        // Emulate vi cursor motions on default configurations.
+        // gg
+        term.scroll_display(Scroll::Top);
+        term.vi_mode_cursor = term.vi_mode_cursor.scroll(&term, 10);
+        // jjjjj
+        for _ in 0..5 {
+            term.vi_mode_cursor = term.vi_mode_cursor.motion(&mut term, ViMotion::Down);
+        }
+        // lll
+        for _ in 0..3 {
+            term.vi_mode_cursor = term.vi_mode_cursor.motion(&mut term, ViMotion::Right);
+        }
+
+        // Clear the viewport.
+        term.clear_screen(ansi::ClearMode::All);
```

Comment:
```suggestion
        // Clear the viewport and scrollback.
        term.clear_screen(ansi::ClearMode::All);
```

Kinda the point that we're not just clearing the viewport.

Reply:
Really? The mode name, `ansi::ClearMode::All` , looks like to clear both. But it only clears the viewport according to the code. In fact, `printf '\033[2J'` clears only the viewport but not the scrollback buffer on Alacritty. Is this my misunderstanding?

Reply:
This situation is caused by `clear -x` , which returns `\033[H\033[2J` . On Alacritty, this changes the cursor position to (1, 1) and clears only the viewport. And, `clear` returns `\033[H\033[2J\033[3J` , which also clears the scrollback buffer, in addition to the previously mentioned behavior.

Reply:
Oh yeah, I'm not sure what I was thinking. We explicitly have `ClearMode::Saved` for the scrollback and the `clear_viewport` method even clearly states that it's only clearing the viewport.

I was just caught up on the bug I suppose: The problem only occurs when the vi mode cursor is in the scrollback and that is "cut off". So I'm not sure your bugfix solves the problem when **only** the saved lines (as in history) is deleted while scrolled up?

Reply:
It's a good point. This problem is caused when clears the scrollback buffer, but it slipped in my mind. I'll fix it later.

Reply:
It causes a freeze, not a crash. This can be fixed by the same way.